Code/src/main/java/nl/utwente/group10/haskell/type/ConstT.java

```diff
@@ -69,4 +70,26 @@ public String toString() {
             return this.constructor;
         }
     }
+
+    @Override
+    public int compareTo(final Type type) {
```

Comment:
Per the Comparable interface, `compareTo` is required to throw a `NullPointerException` if the passed object is `null`. It probably would do just that because you're using `type` later on, but it might be neater to explicitly check for null here.